src/patpy/tl/sample_representation.py

```diff
@@ -1924,6 +1923,253 @@ def calculate_distance_matrix(self, force: bool = False):
         self.sample_representation = distances
         self.samples = list(self.sample_representation.index)
 
-        self.adata.uns[self.DISTANCES_UNS_KEY] = distances
+        self.adata.uns[self.DISTANCES_UNS_KEY] = distances.to_numpy()
+
+        return distances.to_numpy()
+
+
+class GloScope_py(SampleRepresentationMethod):
+    def __init__(self, sample_key, cell_group_key=None, layer="X_pca", seed=67, k=25, use_gpu=False):
+        super().__init__(sample_key=sample_key, cell_group_key=cell_group_key, layer=layer, seed=seed)
+        self.k = k
+        self.use_gpu = use_gpu
+
+        if self.use_gpu:
+            self.DISTANCES_UNS_KEY = "X_gloscope_cuml_distances"
+        else:
+            self.DISTANCES_UNS_KEY = "X_gloscope_pynndescent_distances"
+    
+    def prepare_anndata(self, adata, n_components=10):
+        """
+        Prepares an AnnData object for further usage.
+
+        Parameters
+        ----------
+        adata : AnnData
+            The AnnData object.
+        n_components : int, default: 10
+            Number of components from the embedding to use (default from .obsm['X_pca']).
+        """
+        super().prepare_anndata(adata)
+
+        # Only keep the first n components of the embedding (10 by default)
+        adata.obsm[self.layer] = adata.obsm[self.layer][:, :n_components]   
```

Comment:
Mutating data is not a good practice. Also, please set the default of `n_components` to None, and not subset the data in this case. I would move `n_components` to either init or `calculate_distance_matrix`.

See a comment below on possible implementation